lib/umwelt/parser.ex

```diff
@@ -0,0 +1,46 @@
+defmodule Umwelt.Parser do
+  @moduledoc "Extracts metadata from AST"
+
+  alias Umwelt.Parser
+
+  def read_ast({:error, msg}), do: {:error, msg}
+
+  def read_ast({:ok, code}),
+    do: Code.string_to_quoted(code)
+
+  def parse({:ok, ast}),
+    do: parse(ast, []) |> index()
+
+  def parse({_, _, _} = ast, aliases),
+    do: Parser.Triple.parse(ast, aliases)
```

Comment:
```suggestion
    do: Parser.Macro.parse(ast, aliases)
```
In AST, triple is actually `Macro.t()` type.